aea/channel/oef.py

```diff
@@ -477,7 +474,7 @@ def connect(self) -> None:
         if self._stopped and not self._connected:
             self._stopped = False
             self._core.run_threaded()
-            assert self.channel.connect()
+            assert self.channel.connect(), "Cannot connect to OEFChannel."
```

Comment:
I think we should remove this `assert` statement. eventually. And we should override the OEF SDK agent methods for connection succeeded/failed/terminated.

Reply:
Agreed, will leave for now though.

Reply:
Yes agreed.